### PR TITLE
Support null-safety

### LINKED ISSUE #37: Use collection literals in bit_pattern.dart
... to clear a best practice/lint.

### LINKED ISSUE #39: Any plans to migrate this wonderful library to sound null safety?
Please migrate to sound null safety. 

Thanks.

### PR DESCRIPTION
**Pull request by leynier**
Support null-safety

fix: #37 
resolve: #39

### CODE REVIEW

**Comment by leynier** [comment]
@matanlurey Please, can you help me to understand why test are failed? Thanks in advance.

**Comment by Akash9151** [comment]
Hey, Is there any chance of adding null-safety support to this package soon..?

**Comment by Akash9151** [comment]
@leynier , I want to do calculations with converting int to binary, Can you suggest me any package..?

**Comment by leynier** [comment]
> @leynier , I want to do calculations with converting int to binary, Can you suggest me any package..?

Hi @Akash9151 sorry for the delay in response, in the meanwhile, you can use the fork made by me.

```
binary:
    git: https://github.com/educup/binary.dart.git
```

**Comment by matanlurey** [comment]
Sorry for the delay, I'm willing to accept this PR or I can fork it and finish it for you @leynier.

Feel free to delete the tests that no longer make sense.

**Comment by leynier** [comment]
> Sorry for the delay, I'm willing to accept this PR or I can fork it and finish it for you @leynier.
> 
> Feel free to delete the tests that no longer make sense.

I made a few changes like upgrade dependencies, GH Actions, move to another linter package, and other things. The tests run correctly in my fork.

File: lib/src/_long.dart [hunk 1]
```diff
@@ -13,16 +11,14 @@ extension BinaryLong on int {
   /// Represents the maximum amount of bits supported in a JS VM.
   static const _maxJs = 52;
 
-  @alwaysThrows
-  // ignore: prefer_void_to_null
-  static Null _max52Bits() {
+  static void _max52Bits() {
```
Comment (matanlurey) [inline lib/src/_long.dart:14]: nit: The right way to do this is have a return type of `Never`

File: analysis_options.yaml [hunk 2]
```diff
@@ -1,23 +1 @@
...
-    - avoid_print
-    - await_only_futures
-    - hash_and_equals
-    - package_api_docs
-    - package_names
-    - package_prefixed_library_names
-    - prefer_final_locals
-    - prefer_void_to_null
-    - unnecessary_overrides
+include: package:lints/recommended.yaml
```
Comment (matanlurey) [inline analysis_options.yaml:1]: nit: Thanks. This is actually less strict than the previous settings to a degree.

I've replaced this with `package:matan/strict.yaml`.

File: lib/src/_utils.dart [hunk 3]
```diff
@@ -9,5 +9,5 @@ extension ObjectX on Object {
   /// Casts the current object to type [T].
   ///
   /// In some production compilers, this cast is skipped.
-  T unsafeCast<T>() => this; // ignore: return_of_invalid_type
+  T unsafeCast<T>() => this as T; // ignore: return_of_invalid_type
```
Comment (matanlurey) [inline lib/src/_utils.dart:12]: nit: Thanks!

To keep this functionality, I needed to add the following pragmas:
```
  @dart2js.tryInline
  @pragma('dart2js:as:trust')
```

File: lib/src/bit_pattern.dart [hunk 4]
```diff
@@ -62,7 +62,7 @@ abstract class BitPatternBuilder {
   ///
   /// print(pattern.match(0xD /* 0b1101 */)); // [1]
   /// ```
-  BitPattern<List<int>> build([String name]);
+  BitPattern<List<int>?> build([String? name]);
```
Comment (matanlurey) [inline lib/src/bit_pattern.dart:65]: Thanks. I think we don't want a `BitPattern` with `null`. I've made some changes.

File: lib/src/int.dart [hunk 5]
```diff
@@ -50,7 +50,7 @@ extension BinaryInt on int {
   static const _maxSmiBits = 31;
 
   /// Represents `math.pow(2, 32)`, precomputed.
-  static const _2p32 = 0x100000000;
+  static const _pow2to32 = 0x100000000;
```
Comment (matanlurey) [inline lib/src/int.dart:53]: Thanks! Much better!

File: makefile [hunk 6]
```diff
@@ -0,0 +1,13 @@
+check: get format analyze tests
```
Comment (matanlurey) [inline makefile:1]: Thanks! Good idea

**Review by matanlurey — APPROVED** [review]
Thanks! I've manually merged with a few changes.

**Pull request merged**